### PR TITLE
Add support for borrowing rvalues

### PR DESCRIPTION
**Pull request by gottesmm**
[silgen] Add support for borrowing an RValue.

rdar://33358110

### CODE REVIEW

**Comment by rjmccall** [comment]
+1

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

File: lib/SILGen/RValue.cpp [hunk 1]
```diff
@@ -672,16 +672,26 @@ void RValue::extractElements(SmallVectorImpl<RValue> &elements) && {
...
+  copiedValues.reserve(values.size());
+  for (ManagedValue v : values) {
+    copiedValues.emplace_back(v.copy(SGF, loc));
   }
+  return RValue(std::move(copiedValues), type, elementsToBeAdded);
+}
+
+RValue RValue::borrow(SILGenFunction &SGF, SILLocation loc) const & {
+  assert((isComplete() || isInSpecialState()) &&
+         "can't borrow incomplete rvalue");
```
Comment (rjmccall) [inline lib/SILGen/RValue.cpp]: These assertions should not allow isInSpecialState().  There's no real way to copy or borrow a null or in-context RValue.

Comment (gottesmm) [inline lib/SILGen/RValue.cpp]: Ok. I was just cargo-culting what was there before. I will fix it. Give me a second.

Comment (gottesmm) [inline lib/SILGen/RValue.cpp]: I am fixing this in #11197

**Pull request merged**